sway/tree/container.c

```diff
@@ -120,12 +120,326 @@ struct sway_container *container_create(struct sway_view *view) {
+			list_find(container->current.workspace->floating, container) != -1) {
+		return true;
+	}
+	if (container->scratchpad) {
+		return true;
+	}
+	return false;
+}
+
+void container_update(struct sway_container *con) {
+	struct border_colors *colors = container_get_current_colors(con);
+	list_t *siblings = NULL;
+	enum sway_container_layout layout = L_NONE;
+
+	if (con->current.parent) {
+		siblings = con->current.parent->current.children;
+		layout = con->current.parent->current.layout;
+	} else if (con->current.workspace) {
+		siblings = con->current.workspace->current.tiling;
+		layout = con->current.workspace->current.layout;
+	}
+
+	const float *bottom, *right;
+	bottom = (const float *) colors->child_border;
+	right = (const float *) colors->child_border;
+
+	if (!container_is_current_floating(con) && siblings && siblings->length == 1) {
+		if (layout == L_HORIZ) {
+			right = (const float *) colors->indicator;
+		} else if (layout == L_VERT) {
+			bottom = (const float *) colors->indicator;
+		}
+	}
+
+	wlr_scene_rect_set_color(con->title_bar.border, (const float *) colors->border);
+	wlr_scene_rect_set_color(con->title_bar.background, (const float *) colors->background);
+	
+	if (con->view) {
+		wlr_scene_rect_set_color(con->border.top, (const float *) colors->child_border);
+		wlr_scene_rect_set_color(con->border.bottom, bottom);
+		wlr_scene_rect_set_color(con->border.left, (const float *) colors->child_border);
+		wlr_scene_rect_set_color(con->border.right, right);
+	}
+
+	// only update the titlebar texture is the color has changed
+	if (colors->text != con->title_bar.last_used_title_color) {
+		container_update_title_textures(con);
+		con->title_bar.last_used_title_color = (const float *) colors->text;
+	}
+}
+
+void container_update_itself_and_parents(struct sway_container *con) {
+	container_update(con);
+
+	if (con->current.parent) {
+		container_update_itself_and_parents(con->current.parent);
+	}
+}
+
+static struct wlr_scene_buffer *create_text_scene_node(
+		struct sway_container *con, const float *color,
+		bool pango_markup, char *text) {
+	int scale = 2.; // output->wlr_output->scale;
+	int width = 0;
+	int baseline;
+	int height = config->font_height;
+	if (height == 0) {
+		return NULL;
+	}
+
+	// We must use a non-nil cairo_t for cairo_set_font_options to work.
+	// Therefore, we cannot use cairo_create(NULL).
+	cairo_surface_t *dummy_surface = cairo_image_surface_create(
+			CAIRO_FORMAT_ARGB32, 0, 0);
+	cairo_t *c = cairo_create(dummy_surface);
+	cairo_set_antialias(c, CAIRO_ANTIALIAS_BEST);
+	cairo_font_options_t *fo = cairo_font_options_create();
+	cairo_font_options_set_hint_style(fo, CAIRO_HINT_STYLE_FULL);
+	//if (output->wlr_output->subpixel == WL_OUTPUT_SUBPIXEL_NONE) {
+		cairo_font_options_set_antialias(fo, CAIRO_ANTIALIAS_GRAY);
+	/*} else {
+		cairo_font_options_set_antialias(fo, CAIRO_ANTIALIAS_SUBPIXEL);
+		cairo_font_options_set_subpixel_order(fo,
```

Comment:
Why is this code commented?

Reply:
Because it's not finished. That's point `Buffer node output awareness` in the checklist.